iceoryx_examples/automotive_soa/include/owl/kom/event_publisher.inl

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_PUBLISHER_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_PUBLISHER_INL
+
+#include "owl/kom/event_publisher.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventPublisher<T, EventTransmission::IOX>::EventPublisher(const core::String& service,
+                                                                 const core::String& instance,
+                                                                 const core::String& event) noexcept
+    : m_publisher({service, instance, event}, {HISTORY_CAPACITY, iox::NodeName_t(), NOT_OFFERED_ON_CREATE})
+{
+}
+
+template <typename T>
+inline void EventPublisher<T, EventTransmission::IOX>::Send(const SampleType& userSample) noexcept
+{
+    auto maybeSample = m_publisher.loan();
+
+    if (maybeSample.has_error())
+    {
+        std::cerr << "Error occured during allocation, couldn't send sample!" << std::endl;
```

Comment:
It is a bad API when we do not inform the user in such a case. Could you please return a `bool` here.

Reply:
I agree with you 100%. However, the top-level API is fixed and can't be changed without going into the steering committee.

Reply:
Then I tend to say that this example should not be part of iceoryx. Would it make maybe sense to create a `eclipse-iceoryx/autosar` repository and store it there.

It feels weird and wrong to emphasize things in our example which we do not believe in.

Reply:
@elfenpiff I know what you mean. However, I'm strongly in favour of keeping the code in one repository for the sake of simplicity. What do you think about highlighting that this API is not recommended with a doxygen comment?

Reply:
@budrus What do you think about changing the API of `Send()`? I personally wouldn't mind.

Reply:
Talked to @budrus and a small deviation like that is ok. I changed the code accordingly.